src/pages/academy/xpCalculation/XpCalculation.tsx

```diff
@@ -0,0 +1,249 @@
+import { IconNames } from '@blueprintjs/icons';
+import { GridApi, GridReadyEvent } from 'ag-grid-community';
+import { AgGridReact } from 'ag-grid-react';
+import { useEffect, useState } from 'react';
+import ContentDisplay from 'src/commons/ContentDisplay';
+
+export type StateProps = {
+  allUserXp: string[][] | undefined;
+};
+
+export type DispatchProps = {
+  handleAllUserXpFetch: () => void;
+};
+
+type XpCalculationProps = DispatchProps & StateProps;
+
+type RowData = {
+  name: string;
+  'NUS Net ID': string;
+  'Assessment Xp': string;
+  'Achievement Xp': string;
+};
+
+type PageState = {
+  currPage: number;
+  maxPages: number;
+  rowCountString: string;
+  isBackDisabled: boolean;
+  isForwardDisabled: boolean;
+};
+
+const XpCalculation = ({ allUserXp, handleAllUserXpFetch }: XpCalculationProps) => {
+  const [rowData, setRowData] = useState<RowData[]>([]);
+  const [gridApi, setGridApi] = useState<GridApi>();
+  const [pageState, setPageState] = useState<PageState>({
+    currPage: 1,
+    maxPages: 1,
+    rowCountString: '(none)',
+    isBackDisabled: true,
+    isForwardDisabled: true
+  });
+  // const [filterText, setFilterText] = useState<string>('');
+
+  useEffect(() => {
+    if (allUserXp) {
+      const rowData = allUserXp.map(data => {
+        const [name, nusNetId, assessmentXp, achievementXp] = data;
+        return {
+          name: name ?? '',
+          'NUS Net ID': nusNetId ?? '',
+          'Assessment Xp': assessmentXp ?? '0',
+          'Achievement Xp': achievementXp ?? '0'
+        };
+      });
+      setRowData(rowData);
+    }
+  }, [allUserXp]);
+
+  const columnDefs = [
+    { field: 'name' },
+    { field: 'NUS Net ID' },
+    { field: 'Assessment Xp' },
+    { field: 'Achievement Xp' }
+  ];
+
+  const defaultColDef = {
+    resizable: true,
+    sortable: true
+  };
+
+  const exportCSV = () => {
+    if (gridApi) {
+      gridApi.exportDataAsCsv({
+        fileName: `SA XP Count (${new Date().toISOString()}).csv`,
+        // Explicitly declare exported columns to avoid exporting trash columns
+        columnKeys: ['name', 'NUS Net ID', 'Assessment Xp', 'Achievement Xp']
+      });
+    }
+  };
+
+  // Forcibly resizes columns to fit the width of the datagrid - prevents datagrid
+  // from needing to render a horizontal scrollbar when columns overflow grid width
+  const resizeGrid = () => {
+    if (gridApi) {
+      gridApi.sizeColumnsToFit();
+    }
+  };
+
+  const changePaginationView = (type: string) => {
+    return () => {
+      if (gridApi) {
+        switch (type) {
+          case 'first':
+            return gridApi.paginationGoToFirstPage();
+          case 'prev':
+            return gridApi.paginationGoToPreviousPage();
+          case 'next':
+            return gridApi.paginationGoToNextPage();
+          case 'last':
+            return gridApi.paginationGoToLastPage();
+          default:
+        }
+      }
+    };
+  };
```

Comment:
Why not make `changePaginationView` take in two arguments, `gridApi` and `type`? That way, it can be memoized for performance.